### PR TITLE
feat: add multi resolver to connectivity test for both tcp and dns

### PR DESCRIPTION
**Pull request by 62w71st**
This model can be used to plug in other kinds of resolvers in the future including geneva based ones.

### CODE REVIEW

File: x/connectivity/connectivity.go [hunk 1]
```diff
@@ -58,6 +58,65 @@ func TestResolverPacketConnectivity(ctx context.Context, resolver transport.Pack
...
+
+	for _, answer := range response.Answer {
+		fmt.Printf("%v\n", answer)
+	}
+
+	return nil
+}
+
+func resolve(ctx context.Context, testDomain string, use_tcp bool, c chan net.IP) (net.IP, error) {
+	tcpClient := dns.Client{}
```
Comment (fortuna) [inline x/connectivity/connectivity.go:82]:
```suggestion
	dnsClient := dns.Client{}
```

File: x/connectivity/connectivity.go [hunk 2]
```diff
@@ -58,6 +58,65 @@ func TestResolverPacketConnectivity(ctx context.Context, resolver transport.Pack
...
+	}
+
+	for _, answer := range response.Answer {
+		fmt.Printf("%v\n", answer)
+	}
+
+	return nil
+}
+
+func resolve(ctx context.Context, testDomain string, use_tcp bool, c chan net.IP) (net.IP, error) {
```
[the remark this answers is not in the document]
Comment (62w71st) [reply x/connectivity/connectivity.go]: done.

File: x/connectivity/connectivity.go (hunk 1 above)
Comment (62w71st) [reply x/connectivity/connectivity.go]: done.

File: x/connectivity/connectivity.go
[the remark this answers is not in the document]
Comment (62w71st) [reply x/connectivity/connectivity.go:1]: Moved the modules to x/dns and added support for []net.IP now.

**Review by 62w71st — COMMENTED** [review]
PTAL

**Comment by 62w71st** [comment]
@fortuna is there anything else in this PR that needs to be done?

**Comment by fortuna** [comment]
Obsoleted by https://github.com/Jigsaw-Code/outline-sdk/pull/141 and upcoming PRs based on https://github.com/Jigsaw-Code/outline-sdk/tree/fortuna-dns.

**Pull request closed without merging**